tutorials/13-seasonal-time-series/13_seasonal_time_series.jmd

```diff
@@ -0,0 +1,308 @@
+        color=5,
+        linewidth=4,
+    )
+
+    scatter!(trend_plt, x, y .+ ymax; label=nothing, color=2, legend=:topleft)
+    cyclic_plt = plot(
+        x,
+        cyclic;
+        color=1,
+        label=nothing,
+        alpha=0.2,
+        title="Cyclic effect",
+        xlabel="Time",
+        ylabel="f₂(t)",
+    )
+    return trend_plt, cyclic_plt
+end
+
+chain = sample(decomp_model(x, c, yf, +), NUTS(), 2000)
+yf_samples = predict(decomp_model(x, c, missing, +), chain)
+m, conf = mean_ribbon(yf_samples)
+predictive_plt = plot(
+    t,
+    m .+ yf_max;
+    ribbon=conf,
+    label="Posterior density",
+    title="Posterior decomposition",
+    xlabel="Time",
+    ylabel="f(t)",
+)
+scatter!(predictive_plt, t, yf .+ yf_max; color=2, label="Data", legend=:topleft)
+
+decomp = get_decomposition(decomp_model, x, c, chain)
+decomposed_plt = plot_fit(t, yf, decomp, yf_max)
+combined_plt = plot(predictive_plt, decomposed_plt...; layout=(4, 1), size=(700, 1000))
+```
+
+Inference is successful and the posterior beautifully captures the data.
+We see that the least squares linear fit deviates somewhat from the posterior trend.
+Since our model takes cyclic effects into account separately,
+we get a better estimate of the true overall trend than if we would have just fitted a line.
+But what frequency content did the model identify?
+
+```julia
+function plot_cyclic_features(βsin, βcos)
+    labels = reshape(["freq = $i" for i in freqs], 1, :)
+    colors = collect(freqs)'
+    style = reshape([i <= 10 ? :solid : :dash for i in 1:length(labels)], 1, :)
+    sin_features_plt = density(
+        βsin[:, :, 1];
+        title="Sine features posterior",
+        label=labels,
+        ylabel="Density",
+        xlabel="Weight",
+        color=colors,
+        linestyle=style,
+        legend=nothing,
+    )
+    cos_features_plt = density(
+        βcos[:, :, 1];
+        title="Cosine features posterior",
+        ylabel="Density",
+        xlabel="Weight",
+        label=nothing,
+        color=colors,
+        linestyle=style,
+    )
+
+    return seasonal_features_plt = plot(
+        sin_features_plt,
+        cos_features_plt;
+        layout=(2, 1),
+        size=(800, 600),
+        legend=:outerright,
+    )
+end
+
+βc = group(chain, :βc).value
+plot_cyclic_features(βc[:, begin:num_freqs, :], βc[:, (num_freqs + 1):end, :])
+```
+
+Plotting the posterior over the cyclic features reveals that the model managed to extract the true frequency content.
+
+Since we wrote our model to accept a combining operator, we can easily run the same analysis for a multiplicative model.
+
+```julia
+yg = g.(t) .+ σ * randn(size(t))
+
+prior_samples = sample(decomp_model(t, c, missing, .*), Prior(), 100)
+y_prior_samples = group(prior_samples, :y).value.data[:, :, 1]
+prior_plt = plot(; title="Prior samples")
+plot!(prior_plt, t, y_prior_samples'; label=nothing, color=1, linewidth=1, alpha=0.5)
+scatter!(prior_plt, t, yg; label="Data", color=2)
```

Comment:
As above, this could be simplified to avoid the MCMCChains internals (which users should not have to know about).